Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -92,71 +228,367 @@ void SkylanderPortalWindow::CreateMainWindow()
+  m_element_filter[1]->setText(tr("Magic"));
+  m_element_filter[2]->setText(tr("Water"));
+  m_element_filter[3]->setText(tr("Tech"));
+  m_element_filter[4]->setText(tr("Fire"));
+  m_element_filter[5]->setText(tr("Earth"));
+  m_element_filter[6]->setText(tr("Life"));
+  m_element_filter[7]->setText(tr("Air"));
+  m_element_filter[8]->setText(tr("Undead"));
+  m_element_filter[9]->setText(tr("Other"));
+
+  search_radio_group->setLayout(search_radio_layout);
+  search_filters_layout->addWidget(search_radio_group);
+
+  auto* other_box = new QGroupBox(tr("Other"));
+  auto* other_layout = new QVBoxLayout;
+  connect(m_only_show_collection, &QCheckBox::toggled, this, &SkylanderPortalWindow::RefreshList);
+  other_layout->addWidget(m_only_show_collection);
+  other_box->setLayout(other_layout);
+  search_filters_layout->addWidget(other_box);
+
+  search_filters_layout->addStretch(50);
+
+  search_filters_group->setLayout(search_filters_layout);
+  search_layout->addWidget(search_filters_group);
+
+  search_layout->addWidget(m_skylander_list);
+
+  search_group->setLayout(search_layout);
+  main_layout->addWidget(search_group);
+
+  main_group->setLayout(main_layout);
+
+  return main_group;
+}
+
+void SkylanderPortalWindow::OnPathChanged()
+{
+  m_collection_path = m_path_edit->text();
+  RefreshList();
+}
+
+void SkylanderPortalWindow::SelectPath()
+{
+  QString dir = QDir::toNativeSeparators(DolphinFileDialog::getExistingDirectory(
+      this, tr("Select Skylander Collection"), m_collection_path));
+  if (!dir.isEmpty())
+  {
+    dir += QString::fromStdString("\\");
+    m_path_edit->setText(dir);
+    m_collection_path = dir;
+  }
+  if (m_only_show_collection->isChecked())
+    RefreshList();
+}
+
+void SkylanderPortalWindow::UpdateSelectedVals()
+{
+  const u32 sky_info = m_skylander_list->currentItem()->data(1).toUInt();
+  if (sky_info != 0xFFFFFFFF)
+  {
+    sky_id = sky_info >> 16;
+    sky_var = sky_info & 0xFFFF;
+  }
+}
+
+void SkylanderPortalWindow::RefreshList()
+{
+  int row = m_skylander_list->currentRow();
+  m_skylander_list->clear();
+  if (m_only_show_collection->isChecked())
+  {
+    QDir collection = QDir(m_collection_path);
+    auto& system = Core::System::GetInstance();
+    for (auto file : collection.entryInfoList(QStringList(tr("*.sky"))))
+    {
+      File::IOFile sky_file(file.filePath().toStdString(), "r+b");
+      if (!sky_file)
+      {
+        continue;
+      }
+      std::array<u8, 0x40 * 0x10> file_data;
+      if (!sky_file.ReadBytes(file_data.data(), file_data.size()))
+      {
+        continue;
+      }
+      auto ids = system.GetSkylanderPortal().CalculateIDs(file_data);
+      if (PassesFilter(file.baseName(), ids.first, ids.second))
+      {
+        const uint qvar = (ids.first << 16) | ids.second;
+        QListWidgetItem* skylander = new QListWidgetItem(file.fileName());
+        skylander->setData(1, qvar);
+        m_skylander_list->addItem(skylander);
```

Comment:
Instead of displaying the filename, this should just display the Skylander name based on the ids you calculated earlier